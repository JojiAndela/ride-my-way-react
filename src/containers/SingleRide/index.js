import React, { Component } from 'react';
// import classes from './style.scss';

class SingleRide extends Component {
  render() {
    return(
      <div>
        Got here
      </div>
    )
  }
}

export default SingleRide
