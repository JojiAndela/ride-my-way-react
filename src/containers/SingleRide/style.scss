.SingleRide{
  margin: 5%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  .SingleCard{
    padding: 20px;
    flex: 1;
    border: 1px solid #eee;
    margin: 5px;
    border-radius: 5px;
  }
}

.join {
  padding: 10px;
  color: #fff;
  background-color: #0e83c4;
  border-radius: 5px;
  font-weight: bold;
  font-size: 150%;
  width: 90%;
  margin: 10px 5%;
  cursor: pointer;
  &:hover{
    background-color: #0d68a1;
  }
}
