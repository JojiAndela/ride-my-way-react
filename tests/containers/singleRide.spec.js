import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

// Component to be tested
import SingleRide from '../../src/containers/SingleRide';

const initialState ={
  rides: {
    loading: false,
    rides: [],
    singleRide: null,
    requests: [],
    myRequest: {}
  }
}
const middleware = [thunk];
const mockStore = configureStore(middleware);

let wrapper;
let store;
describe('SingleRide', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <SingleRide match={{params: { rideId: 1 }}}/>
        </BrowserRouter>
      </Provider>
    );
  });
  it('should render the single ride page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
