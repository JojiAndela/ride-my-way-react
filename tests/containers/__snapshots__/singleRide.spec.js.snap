// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleRide should render the single ride page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(SingleRide)
        match={
          Object {
            "params": Object {
              "rideId": 1,
            },
          }
        }
      >
        <SingleRide
          getRequest={[Function]}
          getSingleRide={[Function]}
          joinSingleRide={[Function]}
          loading={false}
          match={
            Object {
              "params": Object {
                "rideId": 1,
              },
            }
          }
          myRequest={Object {}}
          ride={null}
        >
          <div
            className="SingleRide"
          />
        </SingleRide>
      </Connect(SingleRide)>
    </Router>
  </BrowserRouter>
</Provider>
`;
